@xen-orchestra/web-core/lib/components/ui/checkbox/UiCheckbox.vue

```diff
@@ -0,0 +1,429 @@
+    }
+
+    &:checked + .fake-checkbox > .icon {
+      opacity: 1;
+    }
+
+    &:disabled {
+      &.accent--brand {
+        & + .fake-checkbox {
+          cursor: not-allowed;
+          border-color: var(--color-neutral-txt-secondary);
+
+          > .icon {
+            color: var(--color-neutral-txt-secondary);
+          }
+        }
+
+        &:checked + .fake-checkbox,
+        &.indeterminate + .fake-checkbox {
+          border-color: transparent;
+          background-color: var(--color-normal-item-disabled);
+        }
+      }
+
+      &.accent--success {
+        & + .fake-checkbox {
+          cursor: not-allowed;
+          border-color: var(--color-neutral-txt-secondary);
+
+          > .icon {
+            color: var(--color-neutral-txt-secondary);
+          }
+        }
+
+        &:checked + .fake-checkbox,
+        &.indeterminate + .fake-checkbox {
+          border-color: transparent;
+          background-color: var(--color-success-item-disabled);
+        }
+      }
+
+      &.accent--warning {
+        & + .fake-checkbox {
+          cursor: not-allowed;
+          border-color: var(--color-neutral-txt-secondary);
+
+          > .icon {
+            color: var(--color-neutral-txt-secondary);
+          }
+        }
+
+        &:checked + .fake-checkbox,
+        &.indeterminate + .fake-checkbox {
+          border-color: transparent;
+          background-color: var(--color-warning-item-disabled);
+        }
+      }
+
+      &.accent--danger {
+        & + .fake-checkbox {
+          cursor: not-allowed;
+          border-color: var(--color-neutral-txt-secondary);
+
+          > .icon {
+            color: var(--color-neutral-txt-secondary);
+          }
+        }
+
+        &:checked + .fake-checkbox,
+        &.indeterminate + .fake-checkbox {
+          border-color: transparent;
+          background-color: var(--color-danger-item-disabled);
+        }
+      }
+    }
+
+    &:not(:disabled) {
+      &.accent--brand {
+        & + .fake-checkbox {
+          border-color: var(--color-normal-item-base);
+        }
+
+        &.indeterminate + .fake-checkbox {
+          background-color: var(--color-normal-item-base);
+
+          &:hover {
+            border-color: transparent;
+            background-color: var(--color-normal-item-hover);
+          }
+
+          &:active {
+            background-color: var(--color-normal-item-active);
+          }
+        }
+
+        &.indeterminate:hover + .fake-checkbox {
+          border-color: transparent;
+          background-color: var(--color-normal-item-hover);
+        }
+
+        &:hover + .fake-checkbox {
+          border-color: var(--color-normal-item-hover);
+        }
+
+        &:focus-visible + .fake-checkbox {
+          outline: none;
+
+          &::before {
+            content: '';
+            position: absolute;
+            inset: -0.6rem;
+            border: 0.2rem solid var(--color-normal-txt-base);
+            border-radius: 0.4rem;
```

Comment:
Maybe this part can be refactored to avoid repetitions in the other variants, and we can alter only the `border-color` based on the variant.